third_party/move/move-binary-format/src/file_format.rs

```diff
@@ -1817,6 +2095,29 @@ pub enum Bytecode {
     #[gas_type_creation_tier_1 = "field_ty"]
     ImmBorrowFieldGeneric(FieldInstantiationIndex),
 
+    #[group = "variant"]
+    #[static_operands = "[variant_field_inst_idx]"]
+    #[description = r#"
+        Consume the reference to a generic struct at the top of the stack,
+        and provided that the struct is of the given variant, load an immutable
+        reference to the field of the variant.
+
+        Aborts execution if the operand is not of the given variant.
+    "#]
+    #[semantics = r#"
+        stack >> struct_ref
+        if struct_ref is variant_field.variant
+            stack << &(*struct_ref).field(variant_field.field_index)
+        else
+            error
+    "#]
+    #[runtime_check_epilogue = r#"
+        ty_stack >> ty
+        assert ty == &mut struct_ty
+        ty_stack << &mut field_ty
+    "#]
+    ImmBorrowVariantFieldGeneric(VariantFieldInstantiationIndex),
```

Comment:
So we are using test instruction to pair with destruct so that we don't need to implement a jump table instead right?  I think victor suggested that this could have some gas implication as later variant might cost more than the earlier ones as you have more instructions to test.